test/residency/contract/test_explanations_public_seam.py

```
@@ -0,0 +1,92 @@
+REPO_ROOT = Path(__file__).resolve().parents[3]
+PUBLIC_SEAM = REPO_ROOT / "test/residency/contract/explanations_public_seam.cpp"
+INCLUDE_ROOT = REPO_ROOT / "src/cpp/include"
+EXPLANATIONS_HEADER = INCLUDE_ROOT / "lemon/residency/explanations.h"
+EXPLANATIONS_SOURCE = REPO_ROOT / "src/cpp/server/residency/explanations.cpp"
+GENERATED_SOURCE = REPO_ROOT / "src/cpp/server/residency/generated_contract.cpp"
+
+
+def configured_compiler() -> list[str]:
+    configured = shlex.split(os.environ.get("CXX", "c++"))
+    if not configured or shutil.which(configured[0]) is None:
+        raise RuntimeError(f"C++ compiler is unavailable: {configured!r}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)